dbt/adapters/bigquery/connections.py

```diff
@@ -587,6 +587,20 @@ def table_ref(database, schema, table_name):
         dataset_ref = google.cloud.bigquery.DatasetReference(database, schema)
         return google.cloud.bigquery.TableReference(dataset_ref, table_name)
 
+    @staticmethod
+    def routine_ref(database: str, schema: str, routine_name: str) -> google.cloud.bigquery.RoutineReference:
+        """Returns a RoutineReference object referencing a BigQuery routine."""
+        return google.cloud.bigquery.RoutineReference.from_string(f"{database}.{schema}.{routine_name}")
+
+    def get_bq_routine(self, database: str, schema: str, identifier: str) -> google.cloud.bigquery.Routine:
+        """Get a BigQuery routine (UDF) for a schema/model."""
+        conn = self.get_thread_connection()
+        # backwards compatibility: fill in with defaults if not specified
+        database = database or conn.credentials.database
+        schema = schema or conn.credentials.schema
+        routine_ref = self.routine_ref(database, schema, identifier)
+        return conn.handle.get_routine(routine_ref)
```

Comment:
any reason you can't use `info_schema.routines`? If we convert these python functions to SQL macro equivalents, we could publish this materialization as a dbt package (at least for an trial period).
https://cloud.google.com/bigquery/docs/information-schema-routines